go/private/repository_tools.bzl

```diff
@@ -76,23 +30,34 @@ filegroup(
 """
 
 def _go_repository_tools_impl(ctx):
-  go_tool = ctx.path(ctx.attr._go_tool)
-  goroot = go_tool.dirname.dirname
-  gopath = ctx.path('')
-  prefix = "github.com/bazelbuild/rules_go/" + ctx.attr._tools.package
-  src_path = ctx.path(ctx.attr._tools).dirname
-
-  _fetch_repository_tools_deps(ctx, goroot, gopath)
-
-  for t, pkg in [("gazelle", 'gazelle/gazelle'), ("fetch_repo", "fetch_repo")]:
-    ctx.symlink("%s/%s" % (src_path, t), "src/%s/%s" % (prefix, t))
+  # We have to download this directly because the normal version is based on go_repository
+  # and thus requires the gazelle we build in here to generate it's BUILD files
+  # The commit used here should match the one in repositories.bzl
+  ctx.download_and_extract(
+      url = "https://codeload.github.com/golang/tools/zip/3d92dd60033c312e3ae7cac319c792271cf67e37",
+      type = "zip",
+  )
 
-    result = ctx.execute([
-        'env', 'GOROOT=%s' % goroot, 'GOPATH=%s' % gopath,
-        go_tool, "build",
-        "-o", ctx.path("bin/" + t), "%s/%s" % (prefix, pkg)])
-    if result.return_code:
-      fail("failed to build %s: %s" % (t, result.stderr))
+  go_tool = ctx.path(ctx.attr._go_tool)
+  x_tools_path = ctx.path('tools-3d92dd60033c312e3ae7cac319c792271cf67e37')
+  buildtools_path = ctx.path(ctx.attr._buildtools).dirname
+  go_tools_path = ctx.path(ctx.attr._tools).dirname
+  # Build something that looks like a normal GOPATH so go install will work
```

Comment:
Nit: adding a blank line before each of these comments would improve readability.